book.rb

```diff
@@ -0,0 +1,17 @@
+require_relative './item'
+
+class Book < Item
+  attr_accessor :publisher, :cover_state
+
+  def initialize(publish_date, archived, id, publisher, cover_state)
+    super(publish_date, archived, id)
+    @publisher = publisher
+    @cover_state = cover_state
+  end
+
+  def can_be_archived?
+    return true if can_be_archived? == true || @cover_state == 'bad'
```

Comment:
The `can_be_archived?` method is in recursion, the **inner** `can_be_archived?` method and the **outer** are the same. Once I runned the code it gave me an infinite loop. You can call the parent method using `super`.
Like this:

```suggestion
    return true if super == true || @cover_state == 'bad'
```
Please change it!